controllers/perconaservermysql_controller.go

```diff
@@ -99,19 +99,90 @@ func (r *PerconaServerMySQLReconciler) Reconcile(
 		return rr, errors.Wrap(err, "get CR")
 	}
 
+	if cr.ObjectMeta.DeletionTimestamp != nil {
+		return rr, r.applyFinalizers(ctx, cr)
```

Comment:
we shouldn't requeue and return just `ctrl.Result{}` here

Reply:
Right, right.